consensus/dummy/consensus.go

```diff
@@ -311,6 +313,14 @@ func (eng *DummyEngine) verifyBlockFee(
 	if requiredBlockGasCost == nil || !requiredBlockGasCost.IsUint64() {
 		return fmt.Errorf("invalid block gas cost (%d) in apricot phase 4", requiredBlockGasCost)
 	}
+	// In Granite we don't need to verify the block fee
```

Comment:
It seems our verification of `BlockGasCost` and the block fee for various upgrades is somewhat inconsistent. For instance in `Finalize`, we check if we're in apricot phase 4 prior to calling `verifyBlockFee`. Would it make more sense to instead modify that check to be `if rules.IsApricotPhase4 && !rules.IsGranite`? 

https://github.com/ava-labs/coreth/blob/master/consensus/dummy/consensus.go#L393